src/change_trust_asset.js

```diff
@@ -0,0 +1,246 @@
+import clone from 'lodash/clone';
+import padEnd from 'lodash/padEnd';
+import trimEnd from 'lodash/trimEnd';
+import xdr from './generated/stellar-xdr_generated';
+import { Asset } from './asset';
+import { Keypair } from './keypair';
+import { LiquidityPoolFeeV18, getLiquidityPoolId } from './liquidity_pool_id';
+import { StrKey } from './strkey';
+
+/**
+ * ChangeTrustAsset class represents a trustline change to either a liquidity
+ * pool, the native asset (`XLM`) or an issued asset with an asset code / issuer
+ * account ID pair.
+ *
+ * In case of the native asset, the code will represent `XLM` while the issuer
+ * and liquidityPoolParameters will be empty. For an issued asset, the code and
+ * issuer will be valid and liquidityPoolParameters will be empty. For liquidity
+ * pools, the liquidityPoolParameters will be valid and the remaining two fields
+ * will be empty.
+ *
+ * @constructor
+ * @param {string} code - The asset code.
+ * @param {string} issuer - The account ID of the asset issuer.
+ * @param {LiquidityPoolParameters} liquidityPoolParameters – The liquidity pool parameters.
+ * @param {Asset} liquidityPoolParameters.asseta – The first asset in the Pool, it must respect the rule assetA < assetB.
+ * @param {Asset} liquidityPoolParameters.assetB – The second asset in the Pool, it must respect the rule assetA < assetB.
+ * @param {number} liquidityPoolParameters.fee – The liquidity pool fee. For now the only fee supported is `30`.
+ */
+export class ChangeTrustAsset {
+  constructor(code, issuer, liquidityPoolParameters) {
```

Comment:
I think it would be better to get rid of the constructor and and two static methods to create ChangeTrustAsset instances:

`static creditAsset(code, issuer)`
and
`static poolShareAsset(liquidityPoolParameters)`

Reply:
That's a great idea! I'm not very experienced in typescript types so I didn't find a clean way to avoid the code repetition... I'll get back to you after I've improved that.

Reply:
Creating the static methods makes a lot of sense to me! If I remove the constructor though, Javascript will use the default constructor:
```js
constructor() {}
```
Since the class won't throw an error if you call it with an empty constructor, my idea is to leave the constructor as is but stop exposing it in [index.d.ts](https://github.com/stellar/js-stellar-base/blob/714c66b12c4eae4f40dd874f650f75cd248206b9/types/index.d.ts#L67), so typescript users will only have access to the static methods. WDYT?

Reply:
sounds good to me. I don't think there's any nice way to make a constructor private in javascript